### ISSUE TITLE
crash in AI battle with troglodytes

### ISSUE THREAD

**Issue by Povelitelh3** [vcmi/vcmi#3212]
I downloaded today's develop and started the game. I crashed once on the first move. I didn’t pay any attention, I thought it was a random crash. I tried to replay the map few times, but nothing happened. Then I started playing another map, this one, and I was able to reproduce the problem from the save.

The problem occurs when the tan AI attacks the troglodytes

'VCMI 1.4.0.ff6260e5c5dcfa0c67c5f58f0d3eb73464fd9eb3', mingw
```

[2023-Nov-21 14:45:54.390573] ERROR mod - Failed to resolve identifier core:quicksand of type battlefield from mod game
[2023-Nov-21 14:45:54.391012] ERROR global - Disaster happened.
```

[SaveLogs.ZIP](https://github.com/vcmi/vcmi/files/13425839/SaveLogs.ZIP)

**Comment by Povelitelh3** [comment]
this does not depend on the AI, I also had this problem when I attacked monsters.

**Comment by IvanSavenko** [comment]
Will be fixed in next daily build or in builds from here: https://github.com/vcmi/vcmi/pull/3208

**Comment by IvanSavenko** [comment]
Fixed in #3208
